### ISSUE TITLE
extract_format_args assist escapes `\` character in `\n` etc.

### ISSUE THREAD

**Issue by DesmondWillowbrook** [rust-lang/rust-analyzer#16745]
**rust-analyzer version**: rust-analyzer version: 0.3.1860-standalone

**rustc version**: rustc 1.76.0 (07dca489a 2024-02-04)

**code snippet to reproduce**: 
Add the following test in `rust-analyzer/crates/ide-assists/src/handlers/extract_expressions_from_format_string.rs`
```rust
    #[test]
    fn escaped_literals() {
        check_assist(
            extract_expressions_from_format_string,
            &add_macro_decl(
                r#"
fn main() {
    print!("\n {x + 1}$0");
}
            "#,
            ),
            &add_macro_decl(
                r#"
fn main() {
    print!("\n {}"$0, x + 1);
}
            "#,
            ),
        );
    }
```
fails and instead produces
```
fn main() {
    print!("\\n {}"$0, x + 1);
}
```

**Comment by Veykril** [comment]
not sure if this is related to your changes or not but it might so pinging @DropDemBits

**Comment by DropDemBits** [comment]
Looks like `extract_expressions_from_format_string` is using snippet stuff but not `insert_snippet` or `replace_snippet`, so I didn't catch it as part of the assists to migrate to the structured snippet API.

It also looks like `parse_format_exprs` is the one escaping the `\`: https://github.com/rust-lang/rust-analyzer/blob/9f14343f9ee24f53f17492c5f9b653427e2ad15e/crates/ide-db/src/syntax_helpers/format_string_exprs.rs#L82-L84

**Comment by DropDemBits** [comment]
I have some time today, so I'll take this up.
@rustbot claim
